main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -561,45 +562,40 @@ object GenExpression {
 
       case AtomicOp.Is(sym) =>
         val List(exp) = exps
-        // We get the JvmType of the class for tag
-        val classType = JvmOps.getTagClassType(sym)
-        // First we compile the `exp`
+        val taggedType = BackendObjType.Tagged
+
         compileExpr(exp)
-        // We check if the enum is `instanceof` the class
-        mv.visitTypeInsn(INSTANCEOF, classType.name.toInternalName)
+        val ins = {
+          import BytecodeInstructions._
+          CHECKCAST(taggedType.jvmName) ~ GETFIELD(taggedType.NameField) ~
+            pushString(JvmOps.getTagName(sym)) ~ INVOKEVIRTUAL(JavaObject.EqualsMethod)
```

Comment:
Not equals! That is slow. It should be ==, i.e. if_acmp

Reply:
But how do we guarantee that they are in fact equal? then we need to have one instance of each of the tag names right?

Reply:
Check what intern does.

Reply:
Real string comparison is way too slow.